### PR TITLE
bug(core/playback/mpv): jukebox mode under windows - #2767

### PR DESCRIPTION
**Pull request by apkatsikas**
### Closes
https://github.com/navidrome/navidrome/issues/2767

### Description
Fixes Jukebox mode when running under Windows.	

Windows needs to use a named pipe instead of a file in the temp folder for the socket with mpv. See https://mpv.io/manual/master#using-mpv-from-other-programs-or-scripts which states regarding `--input-ipc-server` flag for script control 
`This gives you access to the JSON IPC over unix domain sockets (or named pipes on Windows).` 

See Windows docs on Named Pipes - https://learn.microsoft.com/en-us/windows/win32/ipc/pipe-names

```
The pipe server cannot create a pipe on another computer, so CreateNamedPipe must use a period for the server name, as shown in the following example.

\\.\pipe\PipeName
```

This is also expounded on here - https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-createnamedpipea

### Changes
- Use named pipe for socket path under Windows during mpv playback/Jukebox mode
- Change function name
- Unexport function

Tested regular playback and Jukebox mode via `rest/jukeboxControl` endpoint, on Windows and Linux.

### CODE REVIEW

**Comment by apkatsikas** [comment]
I noticed two errors being emitted when a track ends and goes to next queued during Jukebox play on Windows, that do not seem to affect playback experience, but are probably specific to how Named Pipes are treated in Windows:

- `"Error sending quit command to mpv-ipc socket" error="trying to send command on closed mpv client"` - https://github.com/navidrome/navidrome/blob/6f7b48202e9135378f3696f28b8f49595f8a8f75/core/playback/mpv/track.go#L114 - my guess is Windows is automatically handling the closing of the Named pipe and we don't have to do anything here.
- `"error cleaning up socketfile": " \\.\pipe\mpvsocket\mpv-ctrl-b400e1e2db7708379f89740f8a897905.socket="!!!!Invalid number of arguments in log call!!!!"` - https://github.com/navidrome/navidrome/blob/6f7b48202e9135378f3696f28b8f49595f8a8f75/core/playback/mpv/track.go#L130 - similar guess, I don't think we need to call os.Remove on a Named Pipe, I believe Windows handles this.

Neither of these errors is affecting the user experience (to my knowledge) but it's not great to throw these errors on Windows due to its special behavior with Named Pipes. The question is - do we want to "fix" these issues in this PR or handle them later?

**Comment by deluan** [comment]
Hey, thanks! I'll take a look and get back to you as soon as I can!

**Comment by ms140569** [comment]
The PR looks totally sensible to me, although i haven't compiled it under Windows. @apkatsikas if you tell/point me on how to build on windows I'll doublecheck for you. Windows is not my Hometurf ...

Regarding the Socket close issue my cut is to handle/fix this within this bug since it's so related.

File: ui/src/personal/LastfmScrobbleToggle.js [hunk 1]
```diff
@@ -23,7 +23,9 @@ const Progress = (props) => {
     )
     const callbackUrl = `${window.location.origin}${callbackEndpoint}`
     openedTab.current = openInNewTab(
-      `https://www.last.fm/api/auth/?api_key=${localStorage.getItem('lastfm-apikey')}&cb=${callbackUrl}`,
+      `https://www.last.fm/api/auth/?api_key=${localStorage.getItem(
+        'lastfm-apikey',
+      )}&cb=${callbackUrl}`,
```
Comment (apkatsikas) [inline ui/src/personal/LastfmScrobbleToggle.js:28]: Was required to run prettier before I could push due to git hooks.

File: core/playback/mpv/track.go [hunk 2]
```diff
@@ -104,6 +105,11 @@ func (t *MpvTrack) Pause() {
 }
 
 func (t *MpvTrack) Close() {
+	// Windows automatically handles closing
+	// and cleaning up named pipe
+	if runtime.GOOS == "windows" {
```
Comment (deluan) [inline core/playback/mpv/track.go:110]: It is not a good practice to use `runtime.GOOS` in your files (unless it is a really small check). Usually we create one file for each variation and use build tags to control which version to use.

For an example, see `scanner/metadata/taglib/get_filename.go` and `scanner/metadata/taglib/get_filename_win.go`

In your case I'd have two `Close()` methods in different files, one for windows (do nothing) and the other for all other platforms with the actual code.

Let me know if it makes sense to you or if you want me to do that.

File: ui/src/personal/LastfmScrobbleToggle.js (hunk 1 above)
Comment (deluan) [reply ui/src/personal/LastfmScrobbleToggle.js:28]: Weird, but that's fine.

File: utils/files.go [hunk 3]
```diff
@@ -3,10 +3,17 @@ package utils
...
 	"os"
 	"path/filepath"
+	"runtime"
 
 	"github.com/google/uuid"
 )
 
-func TempFileName(prefix, suffix string) string {
-	return filepath.Join(os.TempDir(), prefix+uuid.NewString()+suffix)
+func RandomSocketOrFileName(prefix, suffix string) string {
```
Comment (deluan) [inline utils/files.go:11]: I rather leave this function (`TempFileName`) alone and create a `SocketName` in move this function into the `core/playback/mpv` package, also selectable by build tags

**Review by deluan — CHANGES_REQUESTED** [review]

**Review by deluan — CHANGES_REQUESTED** [review]
Thanks, see comments below.

File: core/playback/mpv/track.go (hunk 2 above)
Comment (apkatsikas) [reply core/playback/mpv/track.go:110]: yeah was wondering about this and hoping to get feedback on what best practice is for this project. makes sense and thank you for the examples - will give it a shot and report back

File: utils/files.go (hunk 3 above)
Comment (apkatsikas) [reply utils/files.go:11]: sounds good

**Pull request merged**